core/includes/menu.inc

```diff
@@ -1603,8 +1603,9 @@ function _menu_tree_data(&$links, $parents, $depth) {
   $tree = array();
   while ($item = array_pop($links)) {
     // We need to determine if we're on the path to root so we can later build
-    // the correct active trail and breadcrumb.
-    $item['in_active_trail'] = in_array($item['mlid'], $parents);
+    // the correct active trail and breadcrumb. A special case is needed for
+    // links to <front>.
+    $item['in_active_trail'] = in_array($item['mlid'], $parents) || (backdrop_is_front_page() && $item['link_path'] == '<front>');
```

Comment:
This approach looks good to me but the test failures indicate the 'link_path' either doesn't exist or (more likely) only exists some of the time.